specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2021-07-01-preview/loadtestservice.json

```diff
@@ -0,0 +1,2639 @@
+            "description": "Required testRunId, if testId is not provided",
+            "type": "string",
+            "maxLength": 50,
+            "minLength": 2,
+            "pattern": "^[a-z0-9_-]*$"
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "Api version",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The requested app components",
+            "schema": {
+              "$ref": "#/definitions/AppComponentsMap"
+            }
+          },
+          "default": {
+            "description": "***Error Responses: ***\n\n * \"400\", \"InvalidHostOrResourceName\", \"Host name or domain is not valid\" .\n\n  * \"400\", \"RequiredTestOrTestRun\", \"Either test name or test run name is required\"\n\n  * \"404\", \"TestRunNotFound\", \"test run not found with given name\"\n\n * \"404\", \"TestNotFound\", \"Test not found with given name\"\n\n * \"500\", \"InternalServerError\", \"Unable to process the request, please retry\"",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponseBody"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code for specific error that occurred.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AppComponent_GetAppComponent_MaximumSet_Gen": {
+            "$ref": "./examples/AppComponent_GetAppComponent_MaximumSet_Gen.json"
+          },
+          "AppComponent_GetAppComponent_MinimumSet_Gen": {
+            "$ref": "./examples/AppComponent_GetAppComponent_MinimumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/file/{fileId}:validate": {
+      "post": {
+        "tags": [
+          "File"
+        ],
+        "summary": "Validate input file. File name must be a valid URL character ^[a-z0-9_-]*$. File size can't be more than 50 MB.",
+        "operationId": "File_FileValidate",
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "fileId",
+            "description": "Unique identifier for test file, must be a valid URL character ^[a-z0-9_-]*$",
+            "required": true,
+            "type": "string",
+            "maxLength": 50,
+            "minLength": 2,
+            "pattern": "^[a-z0-9_-]*$"
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "Api version",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "formData",
+            "name": "file",
+            "required": true,
+            "type": "file"
+          }
```

Comment:
It is unnecessary and overly complicated to use `formData` to send a single file-type parameter.
This should be defined as a body parameter with `type: string, format: binary` and use `content-type application/octet-stream`.

Reply:
Hello @mikekistler  , We will change this implementation in our next API revision

Reply:
Sounds good ... thx!